modules/modeling/module_modeling_llm/module_modeling_llm/helpers/llm_utils.py

```diff
@@ -145,19 +36,18 @@ async def predict_and_parse(
     if experiment.run_id is not None:
         tags.append(f"run-{experiment.run_id}")
 
-    if supports_function_calling(model):
-        chain = create_structured_output_chain(pydantic_object, llm=model, prompt=chat_prompt, tags=tags)
-
-        try:
-            return await chain.arun(**prompt_input)
-        except (OutputParserException, ValidationError):
-            # In the future, we should probably have some recovery mechanism here (i.e. fix the output with another prompt)
-            return None
+    structured_output_llm = model.with_structured_output(pydantic_object, method="json_mode")
+    chain = RunnableSequence(
+        chat_prompt,
+        structured_output_llm
+    )
 
-    output_parser = PydanticOutputParser(pydantic_object=pydantic_object)
-    chain = LLMChain(llm=model, prompt=chat_prompt, output_parser=output_parser, tags=tags)
     try:
-        return await chain.arun(**prompt_input)
-    except (OutputParserException, ValidationError):
+        return await chain.ainvoke(prompt_input, config={"tags": tags})
+    except (ValidationError) as e:
         # In the future, we should probably have some recovery mechanism here (i.e. fix the output with another prompt)
-        return None
+        # Show the exception
+
+        print("OutputParserException or ValidationError", e)
```

Comment:
With a corresponding import
```suggestion
        logger.error("Exception type: %s, Message: %s", type(e).__name__, e)
```